src/replit/ai/modelfarm/image_generation_model.py

```diff
@@ -0,0 +1,170 @@
+            async with session.post(url, headers=headers, json=params) as response:
+                await self._check_aresponse(response)
+                return ImageGenerationModelResponse(**await response.json())
+
+    def generate(
+        self,
+        prompt: str,
+        model_name: str = "stabilityai/stable-diffusion-xl-base-1.0",
+        negative_prompt: str = "",
+        loras: list[LoraWeight] = [],
+        seed: Optional[int] = None,
+        image_size: ImageSizeInput = "square_hd",
+        num_inference_steps: int = 30,
+        guidance_scale: float = 7.5,
+        clip_skip: int = 0,
+        model_architecture: Optional[Literal["sd", "sdxl"]] = None,
+        scheduler: Optional[Scheduler] = None,
+        image_format: Literal["jpeg", "png"] = "png",
+        num_images: int = 1,
+    ):
+        """
+        Generate an image from a text prompt.
+
+        Parameters:
+            prompt (str): The prompt to use for generating the image
+            model_name (str): URL or HuggingFace ID of the base model to generate the image
+            negative_prompt (str): Prompt for details that you don't want in the image
+            loras (list[LoraWeight]): The LoRAs to use for the image generation
+            seed (int): Same seed and prompt given to a model will result in same image
+            image_size (ImageSizeInput): The size of the generated image
+            num_inference_steps (int): Number of inference steps
+            guidance_scale (float): Guidance scale (CFG)
+            clip_skip (int): Skips part of the image generation process
+            model_architecture (str): The architecture of the model to use, "sd" or "sdxl"
+            scheduler (str): Scheduler / sampler to use for the image denoising process
+            image_format (str): The format of the generated image, "png" or "jpeg"
+            num_images (int): Number of images to generate in one request
+
+        Returns:
+            ImageGenerationModelResponse
+        """
+        url = f"{self.model_url}"
+        headers = {"Authorization": f"Basic {self.fal_key_id}:{self.fal_key_secret}"}
+        response = requests.post(
+            url,
+            headers=headers,
+            json={
+                "prompt": prompt,
+                "model_name": model_name,
+                "negative_prompt": negative_prompt,
+                "loras": loras,
+                "seed": seed,
+                "image_size": image_size,
+                "num_inference_steps": num_inference_steps,
+                "guidance_scale": guidance_scale,
+                "clip_skip": clip_skip,
+                "model_architecture": model_architecture,
+                "scheduler": scheduler,
+                "image_format": image_format,
+                "num_images": num_images,
+            },
+        )
+        self._check_response(response)
+        return ImageGenerationModelResponse(**response.json())
```

Comment:
would it be better if the sync solution just calls the async one and waits for it to finishi? it would be like the 

```
loop = asyncio.get_running_loop()
await loop.run_in_executor(...)
```

we have done internally.


No hard opinions, just an idea.